### ISSUE TITLE
DataDog is showing weird metric names (behaviour change)

### ISSUE THREAD

**Issue by Arkatufus** [AppMetrics/AppMetrics#682]
Version: 4.2.0
Reporter used: App.Metrics.Datadog
.NET Core version used: netcoreapp3.1

Datadog dashboard is showing different metric names than it used to for Meter metric types. Before it was `Akka.NET.akka.messages.recv.meter.value`, and now it changes to `Akka.NET.akka.messages.recv.meter.count.meter`, breaking previously working DataDog dashboards.

Screenshot:
![image](https://user-images.githubusercontent.com/8293237/117220839-3ecaad80-ae32-11eb-82ee-0493c28af752.png)

**Comment by Aaronontheweb** [comment]
@Arkatufus this broke between 4.1 and 4.2 right?

**Comment by Arkatufus** [comment]
No, it has always been broken on App.Metrics side, the reporting formatter is different compared to what DataDog agent uses

**Comment by Arkatufus** [comment]
In essence, if you used a StatsD client to submit metrics to DataDog StatsD Agent, and then changed and use App.Metrics.Reporting.DataDog to submit metrics via DataDog HTTP client, all your old metrics and your dashboard will be hosed because the incompatible metrics formatting.

**Comment by Arkatufus** [comment]
These code have to modified so that they reflect how DataDog StatsD Agent formats each metric types:
https://github.com/AppMetrics/AppMetrics/blob/features/4.2.0/src/Reporting/src/App.Metrics.Formatting.Datadog/DefaultDatadogMetricJsonWriter.cs#L35-L95
